### PR TITLE
pool: Support TLS

### PR DESCRIPTION
**Pull request by jianghushinian**
server support TLS

### CODE REVIEW

File: pool/pool_server.py [hunk 1]
```diff
@@ -48,12 +51,27 @@ def check_authentication_token(launcher_id: bytes32, token: uint64, timeout: uin
...
+    return ssl_context
+
+
 class PoolServer:
     def __init__(self, config: Dict, constants: ConsensusConstants, pool_store: Optional[AbstractPoolStore] = None):
 
         self.log = logging.getLogger(__name__)
         self.pool = Pool(config, constants, pool_store)
 
+        with open(os.getcwd() + "/config.yaml") as f:
```
Comment (xdustinface) [inline pool/pool_server.py:68]: Hm, there are two places now where this happens, lets wrap it or pass it in here?

Comment (jianghushinian) [reply pool/pool_server.py:66]: good suggestion, pass it in here.

**Review by xdustinface — CHANGES_REQUESTED** [review]
Can you please drop all unrelated changes like formatting, remove unused, ... ?

**Comment by jianghushinian** [comment]
> Can you please drop all unrelated changes like formatting, remove unused, ... ?

It was deleted.

**Review by xdustinface — APPROVED** [review]

**Comment by jack60612** [comment]
Seems good!

**Review by mariano54 — APPROVED** [review]

**Pull request merged**